### ISSUE TITLE
Interface & implementation expects float, but database returns string

### ISSUE THREAD

**Issue by peterjaap** [yireo/Yireo_GoogleTagManager2#78]
We got this error on a client shop:

```
2021/07/13 12:16:26 [error] 31476#31476: *16253478 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Declaration of Yireo\GoogleTagManager2\ViewModel\Product::getProductPrice(Magento\Catalog\Api\Data\ProductInterface $product) must be compatible with Yireo\GoogleTagManager2\Api\ProductViewModelInterface::getProductPrice(Magento\Catalog\Api\Data\ProductInterface $product): float in /data/web/releases/20210713102022/vendor/yireo/magento2-googletagmanager2/ViewModel/Product.php on line 23" while reading response header from upstream, client: 66.249.92.59, request: "GET /en/product-url-here HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000"
```

In these files, a return type hint has been added;

https://github.com/yireo/Yireo_GoogleTagManager2/blob/master/Api/ProductViewModelInterface.php#L30

https://github.com/yireo/Yireo_GoogleTagManager2/blob/master/ViewModel/Product.php#L95

When a catalog price rule has been configured, the result from `getFinalPrice()` is a string, and not a float, causing a 500 error.

**Comment by jissereitsma** [comment]
Thanks, this should now be fixed in 2.1.1.

**Comment by peterjaap** [comment]
:+1:
